itest/rfq_test.go

```diff
@@ -214,6 +215,164 @@ func testRfqAssetBuyHtlcIntercept(t *harnessTest) {
+		require.True(t.t, ok, "unexpected event: %v", event)
+
+		return nil
+	}, defaultWaitTimeout)
+	require.NoError(t.t, waitErr)
+
+	// Alice should have received an accepted quote from Bob. This accepted
+	// quote can be used by Alice to make a payment to Bob.
+	acceptedQuotes, err := ts.AliceTapd.QueryPeerAcceptedQuotes(
+		ctxt, &rfqrpc.QueryPeerAcceptedQuotesRequest{},
+	)
+	require.NoError(t.t, err, "unable to query accepted quotes")
+	require.Len(t.t, acceptedQuotes.SellQuotes, 1)
+
+	acceptedQuote := acceptedQuotes.SellQuotes[0]
+
+	// Register to receive RFQ events from Bob's tapd node. We'll use this
+	// to wait for Bob to receive the HTLC with the asset transfer specific
+	// scid.
+	bobEventNtfns, err := ts.BobTapd.SubscribeRfqEventNtfns(
+		ctxb, &rfqrpc.SubscribeRfqEventNtfnsRequest{},
+	)
+	require.NoError(t.t, err)
+
+	// Carol generates and invoice for Alice to settle.
+	addInvoiceResp := ts.CarolLnd.RPC.AddInvoice(&lnrpc.Invoice{
+		ValueMsat: int64(askAmt),
+	})
+	invoice := ts.CarolLnd.RPC.LookupInvoice(addInvoiceResp.RHash)
+
+	// We now need to construct a route for the payment from Alice to Carol.
+	// The route will be Alice -> Bob -> Carol. We'll add the accepted quote
+	// ID as a record to the custom records field of the route's first hop.
+	// This will allow Bob to validate the payment against the accepted
+	// quote.
+	routeBuildRequest := routerrpc.BuildRouteRequest{
+		AmtMsat: int64(askAmt),
+		HopPubkeys: [][]byte{
+			ts.BobLnd.PubKey[:],
+			ts.CarolLnd.PubKey[:],
+		},
+	}
+	routeBuildResp := ts.AliceLnd.RPC.BuildRoute(&routeBuildRequest)
+
+	// Add the accepted quote ID as a record to the custom records field of
+	// the route's first hop.
+	aliceBobHop := routeBuildResp.Route.Hops[0]
+	if aliceBobHop.CustomRecords == nil {
+		aliceBobHop.CustomRecords = make(map[uint64][]byte)
+	}
+
+	aliceBobHop.CustomRecords[rfq.LnCustomRecordType] =
+		acceptedQuote.Id[:]
+
+	// Update the route with the modified first hop.
+	routeBuildResp.Route.Hops[0] = aliceBobHop
+
+	// Send the payment to the route.
+	t.Log("Alice paying invoice")
+	ts.AliceLnd.RPC.SendToRouteV2(&routerrpc.SendToRouteRequest{
+		PaymentHash: invoice.RHash,
+		Route:       routeBuildResp.Route,
+	})
+
+	// At this point Bob should have received a HTLC with the asset transfer
+	// specific scid. We'll wait for Bob to publish an accept HTLC event and
+	// then validate it against the accepted quote.
+	waitErr = wait.NoError(func() error {
+		t.Log("Waiting for Bob to receive HTLC")
+
+		event, err := bobEventNtfns.Recv()
+		require.NoError(t.t, err)
+
+		acceptHtlc, ok := event.Event.(*rfqrpc.RfqEvent_AcceptHtlc)
+		if ok {
+			require.Equal(
+				t.t, acceptedQuote.Scid,
+				acceptHtlc.AcceptHtlc.Scid,
+			)
+			t.Log("Bob has accepted the HTLC")
+			return nil
+		}
+
+		return fmt.Errorf("unexpected event: %v", event)
+	}, defaultWaitTimeout)
+	require.NoError(t.t, waitErr)
+
+	// TODO(ffranr): Confirm that Carol receives the lightning payment from
```

Comment:
Can we implement this TODO in this PR? Or anything still missing?

Reply:
Could just assert the invoice here?

Reply:
hmm I can't get this to work for some reason. Route is: `Alice -> Bob -> Carol`, the HTLC does seem to reach Bob. And Bob does accept the intercepted HTLC. But then the send function returns:
```
Failure_INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS (1)
```

And so when i lookup the invoice from Carol, it isn't settled.

Reply:
I've attempted to disable Bob's tap HTLC interception but the payment still doesn't succeed. Any ideas @guggero @GeorgeTsagk ?

Reply:
I've got the payment to succeed.